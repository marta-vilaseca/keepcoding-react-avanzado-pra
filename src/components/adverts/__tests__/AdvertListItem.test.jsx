import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AdvertListItem from "../AdvertListItem";

const advert = {
  id: "1",
  name: "Example Advert",
  price: 100,
  sale: true,
  photo: "example-photo.jpg",
  tags: ["tag1", "tag2"],
};

test("renders AdvertListItem correctly", () => {
  const { container } = render(
    <MemoryRouter>
      <AdvertListItem {...advert} />
    </MemoryRouter>
  );

  const snapshot = container.innerHTML;

  expect(snapshot).toMatchSnapshot(`
    "<li class="advert__card"><span class="card__price">100â‚¬</span><div class="card__info"><span class="card__sale">For sale</span><h3 class="card__title"><a href="/adverts/1">Example Advert</a></h3></div><div class="card__photo"><a href="/adverts/1"><img alt="Example Advert" src="example-photo.jpg"></a></div><ul class="card__tags"><li class="card__tag">tag1</li><li class="card__tag">tag2</li></ul></li>"
  `);
});
