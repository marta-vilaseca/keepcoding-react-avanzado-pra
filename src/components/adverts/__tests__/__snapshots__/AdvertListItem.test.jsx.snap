// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders AdvertListItem correctly > 
    "<li class="advert__card"><span class="card__price">100€</span><div class="card__info"><span class="card__sale">For sale</span><h3 class="card__title"><a href="/adverts/1">Example Advert</a></h3></div><div class="card__photo"><a href="/adverts/1"><img alt="Example Advert" src="example-photo.jpg"></a></div><ul class="card__tags"><li class="card__tag">tag1</li><li class="card__tag">tag2</li></ul></li>"
   1`] = `"<li class="advert__card"><span class="card__price">100€</span><div class="card__info"><span class="card__sale">For sale</span><h3 class="card__title"><a href="/adverts/1">Example Advert</a></h3></div><div class="card__photo"><a href="/adverts/1"><img alt="Example Advert" src="example-photo.jpg"></a></div><ul class="card__tags"><li class="card__tag">tag1</li><li class="card__tag">tag2</li></ul></li>"`;
